tools/data_renewal_generate.py

```diff
@@ -0,0 +1,149 @@
+import os
+import csv
+import yaml
+import json
+import sys
+
+from yaml.loader import FullLoader
+
+write_header = True
+total_count = 0
+
+# list of values that we care about
+_KEY_FILTER = [
+    "type",
+    "description",
+    "bugs",
+    "data_reviews",
+    "data_sensitivity",
+    "notification_emails",
+    "notification",
+    "expires",
+]
+
+def response(last_key, content, expire_version, writer, renewal):
+    global write_header
+    global total_count
+    for key, value in content.items():
+        if (key == "$schema") or (key == "no_lint"):
+            continue
+
+        if ("expires" in value) and ((value["expires"] == "never") or (not value["expires"] <= expire_version)):
+            continue
+
+        if (key == "type"):
+            remove_keys = []
+            for key in content.keys():
+                if (key not in _KEY_FILTER):
+                    remove_keys.append(key)
+
+            for key in remove_keys:
+                content.pop(key)
+
+            total_count += 1
+
+            # name of the telemtry
+            result = {"#": total_count, "name" : last_key.lstrip('.')}
+            result.update(content)
+
+            # add columns for product to fille out
+            result.update({"keep(Y/N)" : ""})
+            result.update({"new expiry version" : ""})
+            result.update({"reason to extend" : ""})
```

Comment:
Why are these updates instead of added to the `result` binding when it's created a few lines above?

Reply:
These are purposely added at the end of result so when the csv file is created, the last 3 columns are these.  These columns are for product to fill in after reviewing.

Reply:
I'll add some comments here to explain, thanks